### ISSUE TITLE
RequestException during Add Integration --> resolved username/password issue

### ISSUE THREAD

**Issue by RockKrawler** [evercape/hass-resol-KM2#9]
**Describe the bug**
After the HACS repository installation and during the Step 1 - Connection Details process, I receive the following error message:
**Failed to Connect**

The following errors were found in the "Log Viewer":
2024-12-30 23:33:01.802 WARNING (SyncWorker_5) [custom_components.resol] RequestException in fetch_data_km2(): Error while fetching data from 192.168.107.9: Expecting value: line 1 column 1 (char 0) Please log any issues here: https://github.com/evercape/hass-resol-KM2/issues
2024-12-30 23:33:01.803 ERROR (MainThread) [custom_components.resol] Failed to connect to Resol device: RequestException in fetch_data_km2(): Error while fetching data from 192.168.107.9: Expecting value: line 1 column 1 (char 0) Please log any issues here: https://github.com/evercape/hass-resol-KM2/issues

From a debug log, I found and ran this:
http://192.168.107.9:80/cgi-bin/get_resol_device_information

**which returned this:**
vendor = "RESOL"
product = "KM2"
serial = "xxxxxxxxxxxxxx"
version = "2.0.6"
build = "rc1"
name = "Resol-KM2"
features = "vbus,wifi,vpn"

and also ran this:
http://192.168.107.9:80/cgi-bin/resol-webservice

**which returned this:**
{
    "error": {
        "category": "App",
        "message": "No path info given"
    }
}

I'm using a Resol KM2 with firmware version 2.0.6.

Please let me know if you need more information.

**Comment by evercape** [comment]
Has it worked before? Which version of HA you using? latest?

I only have KM2 with firmware 1.2.0 to test, but unless Resol dramatically changed the devices with newer firmware this should work.

Is there any chance that you could provide me with access to your KM2 to test? Public IP or DynDns with port forwarding?

Pls switch on full logging in your customise.yaml file and send full logs filtered over "Resol".
Thanks
Martin

**Comment by RockKrawler** [comment]
This is the first install so no, it has not worked in this environment before.

Home Assistant version information is below:
Core - 2024.12.5
Supervisor - 2024.12.0
Operating System - 14.1
Frontend - 20241127.8

I'll work on the customize.yaml file this afternoon.  I don't currently have a customize.yaml, could you provide what I need to include in it for full logging?

**Comment by evercape** [comment]
sorry I meant configuration.yaml

logger:
  default: warn
  logs:
    custom_components.resol: debug

that should give you detailed logs.

ideally if you can provide me with remote access to the IP of the resol, and username/password.

Have a good start into 2025.

**Comment by RockKrawler** [comment]
I'm still working on whether we can give the remote access.  My admin is not too excited about that.

Here are the logs entries after a restart of HA and attempt to add integration:
----------------------------------------------------------------------------
Resol Controller
Version: 2024.11.0
Domain: resol
If you have any issues with this custom component please open an issue here:
https://github.com/evercape/hass-resol-KM2/issues
----------------------------------------------------------------------------
2024-12-31 12:45:05.765 DEBUG (SyncWorker_3) [custom_components.resol] Attempting to discover Resol device via get: http://192.168.107.9:80/cgi-bin/get_resol_device_information
2024-12-31 12:45:05.767 DEBUG (SyncWorker_3) [custom_components.resol] 881E5901BE3C: Resol device data received: {'product': 'KM2', 'vendor': 'RESOL', 'serial': '881E5901BE3C', 'version': '2.0.6', 'build': 'rc1', 'name': 'Resol-KM2', 'features': 'vbus,wifi,vpn', 'host': '192.168.107.9', 'port': '80', 'mac': '88:1E:59:01:BE:3C'}
2024-12-31 12:45:05.770 DEBUG (SyncWorker_0) [custom_components.resol] 881E5901BE3C: KM2 requesting sensor data from url http://192.168.107.9:80/cgi-bin/resol-webservice
2024-12-31 12:45:05.820 WARNING (SyncWorker_0) [custom_components.resol] RequestException in fetch_data_km2(): Error while fetching data from 192.168.107.9: Expecting value: line 1 column 1 (char 0) Please log any issues here: https://github.com/evercape/hass-resol-KM2/issues
2024-12-31 12:45:05.822 ERROR (MainThread) [custom_components.resol] Failed to connect to Resol device: RequestException in fetch_data_km2(): Error while fetching data from 192.168.107.9: Expecting value: line 1 column 1 (char 0) Please log any issues here: https://github.com/evercape/hass-resol-KM2/issues

Thank you so much for the help ...............

**Comment by evercape** [comment]
I think you may have gotten the wrong username/password.

the device data is available without authentication.

the other data requires login username/password.

maybe double check that.

**Comment by RockKrawler** [comment]
Ok, when I try the original default password from the sticker from the KM2 box, I get this:
2024-12-31 13:10:19.105 DEBUG (SyncWorker_2) [custom_components.resol] Attempting to discover Resol device via get: http://192.168.107.9:80/cgi-bin/get_resol_device_information
2024-12-31 13:10:19.105 DEBUG (SyncWorker_2) [custom_components.resol] 881E5901BE3C: Resol device data received: {'product': 'KM2', 'vendor': 'RESOL', 'serial': '881E5901BE3C', 'version': '2.0.6', 'build': 'rc1', 'name': 'Resol-KM2', 'features': 'vbus,wifi,vpn', 'host': '192.168.107.9', 'port': '80', 'mac': '88:1E:59:01:BE:3C'}
2024-12-31 13:10:19.107 DEBUG (SyncWorker_4) [custom_components.resol] 881E5901BE3C: KM2 requesting sensor data from url http://192.168.107.9:80/cgi-bin/resol-webservice
2024-12-31 13:10:19.165 WARNING (SyncWorker_4) [custom_components.resol] 881E5901BE3C: Authentication failed: Invalid credentials
2024-12-31 13:10:19.165 ERROR (MainThread) [custom_components.resol] Authentication failed: Invalid credentials Please log any issues here: https://github.com/evercape/hass-resol-KM2/issues

Since I don't get authentication failed errors with the current password, I assumed authentication was correct.  I do question the "username" as a required field because I do not need that when connecting to the KM2 via a browser.  Is that a possibility?

**Comment by RockKrawler** [comment]
Ok, so I'm in ......

All I did was change the password back to the default.

Password that does work:  9HweSJeb
Password that does not work:  &*J\P>tl*p>6[P=[veVa

Can you tell me which characters are not valid with the Resol Controller integration so I can change it from the default?

**Comment by evercape** [comment]
thanks - so basically it was an issue with username/password?

I am just using normal password authentication. 
I think any piping and & is not a good idea, as the resol does http authentication as well. so maybe only use upper/lower case and numbers. 

I think the limitation is on Resol side.

Pls confirm its all working now?

Thanks
Martin

**Comment by RockKrawler** [comment]
Yes sir, the problem was a password issue.  The symbols in the password worked well when accessing the KM2 via a browser but not so much when using the integration.  All is well now and I did use your recommendation for upper/lower case and numbers only.

Thank you so much for the help in diagnosing my issue and even more so in developing this HA integration to begin with.  So very much appreciated.

Thanks again and hope you have a great 2025.

**Comment by evercape** [comment]
Thanks, glad it's working. All the best for 2025.
